src/core/Akka.Cluster/ClusterHeartbeat.cs

```diff
@@ -742,12 +749,10 @@ internal class RingComparer : IComparer<UniqueAddress>
             /// <inheritdoc/>
             public int Compare(UniqueAddress x, UniqueAddress y)
             {
-                var result = Member.AddressOrdering.Compare(x.Address, y.Address);
-                if (result == 0)
-                    if (x.Uid < y.Uid) return -1;
-                    else if (x.Uid == y.Uid) return 0;
-                    else return 1;
-                return result;
+                var ha = x.Uid;
```

Comment:
Shifts the ring to primarily factor in the UID in the node ring comparisons, rather than the address. Cheaper and faster while remaining stable so long as the membership doesn't change.